src/web/app/Page.re

```diff
@@ -236,6 +328,88 @@ module Update = {
         ...model,
         explain_this,
       };
+    | Assistant(action) =>
```

Comment:
there's a lot of logic below. this should be either somewhere else or broken out into its own file or possibly some helpers. i'm having a hard time following exactly what's happening here

Reply:
dependency injection is possibly a way to go... which is the problematic reference?